expression/builtin_compare_vec.go

```diff
@@ -779,11 +780,47 @@ func (b *builtinLEDurationSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colu
 }
 
 func (b *builtinCoalesceJSONSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCoalesceJSONSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var buf *chunk.Column
+	var err error
+	n := input.NumRows()
+	res := make(map[int]*json.BinaryJSON)
+	result.ReserveJSON(n)
+	var count int
+	for _, a := range b.getArgs() {
+		buf, err = b.bufAllocator.get(types.ETJson, n)
+		if err != nil {
+			return err
+		}
+		defer b.bufAllocator.put(buf)
```

Comment:
`defer` does not execute until the function return. So in this case, there will alloc `len(b.getArgs()) `  columns and will be gc when this function return?